vault/resource_database_secret_backend_connection.go

```diff
@@ -259,7 +259,7 @@ func databaseSecretBackendConnectionResource() *schema.Resource {
 				Type:          schema.TypeList,
 				Optional:      true,
 				Description:   "Connection parameters for the postgresql-database-plugin plugin.",
-				Elem:          connectionStringResource(),
```

Comment:
@DesmondH0 Looking good! As per your question to the community, after the updates added in 1.8+, all databases except for HanaDB support username templating https://www.vaultproject.io/docs/secrets/databases#database-capabilities 

That said I think we can add the `username_template` field into the `connectionStringResource` function itself, and create a separate `connectionStringResourceWithoutUsernameTemplate` function that will only be used by HanaDB

Reply:
Thanks for answering this, I had updated the PR as you described
also updating the .md document for the database which is using `connectionStringResource`

And tested against with my postgres, it looks good to me